gulpHelpers.js

```diff
@@ -80,11 +86,11 @@ module.exports = {
 
     return modules;
   },
-  getModules: _.memoize(function(externalModules) {
+  getModules: _.memoize(function(externalModules, mPath) {
```

Comment:
since `mPath` is not used I  imagine this is for some custom integration? I see two potential problems:

 - `getModules` is used before (and in preparation for) the build, so if `mPath` referes to for example `build/dist` it may not exist;
 - `_.memoize` could cause issues if `mPath` is passed in only sometimes.

Reply:
Actually make it 3+ - the build output directory (e.g. `build/dist`) contains more than just modules; depending on when exactly this is run, it will also have:

 - libraries
 - chunk-core.js and prebid-core.js
 - metadata files for each module (added in 10)

Reply:
Sorry - I accidentally edited your comment when I meant to reply. Here's my reply again, I'll see if I can restore your original comment.

> `mPath` is used below on line 93.

I mean that nothing (or at least nothing in this codebase) passes in `mPath`. I see you just added logic that passes it to `getModuleNames`, which is a separate function that does not take it. 

Even fixing that, all the issues above still apply. Would you be able to to provide steps to reproduce the issue this is attempting to fix?

Reply:
@dgirardi Just out of curiosity, i executed command
`gulp build --modules=pubmaticBidAdapter,pubmaticAnalyticsAdapter`
`gulp bundle --modules=pubmaticBidAdapter,appnexusBidAdapter`

instead of throwing error `Invalid module name: appnexusBidAdapter`
it threw an exception
`Error: File not found`
Hence i thought of fixing it to get the modules from the desired location and not always from `modules/` folder

Reply:
I'd argue that `File not found` make more sense - `appnexusBidAdapter` is /not/ an invalid module, it just wasn't built. And if (ignoring why it doesn't work now) we fix it in the way this PR intend, you'd get things like `prebid` or `ortbConverter` as valid modules, which they are not.

Reply:
I agree that the error message could be more helpful, but I do not agree that we should look for modules someplace where there are not (only) modules. It's just incorrect. Consider that what this PR appears to intend would concatenate the complete bundle (e.g. `prebid.js`) back into the bundle if you pass no module list.